docs/concept-workflows.md

```diff
@@ -1,30 +1,210 @@
+
+#### Non-retryable error reasons
+
+- **Description**: Specifies errors that shouldn't be retried.
+- **Use-case**: There may be errors that you know of that should not trigger a retry.
+In this case you can specify them such that if they occur, the Workflow will not be retried.
+
+### The Task Queue
+
+The only required Workflow options parameter is the name of a [Task Queue](/docs/concept-task-queues).
+Read the [Task Queues concept page](/docs/concept-task-queues) for a better overview.
+Essentially a Task Queue is the mechanism by which any given Worker knows which piece of code to execute next.
+A Workflow can only use one Task Queue, and the from a developer's perspective it is named and managed as a simple string value.
+
+### Custom Id
+
+You may assign a custom Id to a Workflow.
+This Id is meant for business level identification such as a customer Id or an order Id.
+The Temporal Server enforces the uniqueness of the Id, within a [Namespace](/docs/server-namespaces) based on the Workflow Id reuse policy.
+Any attempt to start a Workflow that has the same Id of a Workflow with a reuse policy that does not allow it, is going to fail with a "Workflow execution already started" error.
+
+:::note
+
+A Workflow is uniquely identified by its Namespace, Workflow Id, and Run Id.
+
+:::
+
+#### Allow duplicate failed only policy
+
+- **Description**: Specifying this means that the Workflow is allowed to start only if a previously executed Workflow with the same Id has failed.
+- **Use case**:
+
+#### Allow duplicate policy
+
+- **Description**: Specifying this means that the Workflow is allowed to start independently of a previous Workflow with the same Id regardless of the its completion status.
+This is the default policy, if one is not specified.
+- **Use case**:
+
+#### Reject duplicate policy
+
+- **Description**: Specifying this means that no other Workflow is allowed to start using the same Workflow Id at all.
+- **Use case**:
+
+## Child Workflows
+
+If a Workflow is started by another Workflow, then it is considered a Child Workflow.
+The completion, or failure, of a Child Workflow is reported to the Workflow that started it (the Parent Workflow).
+
+The following is a list of some of the more common reasons why you might want to break up code execution into Child Workflows:
+
+- Execute code using a different set of Workers.
+- Enable invocation from multiple other Workflow.
+- Workaround event history size limits.
+- Create one-to-one mappings between a Workflow Id and some other resource.
+- Execute some periodic logic.
+
+One of the main reasons you would not want to use a Child Workflow is the lack of a shared state with the Parent Workflow.
+A Parent and Child Workflow can communicate only through asynchronous signals.
+If the executing logic has tight coupling between Workflows, it may simply be easier to use a single Workflow that can rely on a shared object's state.
+
+## Example subscription Workflow use-case
```

Comment:
I don't like that we moved the example to the end. I think we should sell the model first without giving too many technical details. Maybe break it into multiple pages if this one is too long.

After the changes this page looks scary to any novice without explaining the value of the platform.